plugins/recover.c

```diff
@@ -216,32 +217,32 @@ static struct command_result *after_listpeerchannels(struct command *cmd,
 	}
 
 	lost_state_timer =
-	    plugin_timer(plugin, time_from_sec(CHECK_PEER_INTERVAL),
+	    global_timer(plugin, time_from_sec(CHECK_PEER_INTERVAL),
 			 do_check_lost_peer, NULL);
 	return command_still_pending(cmd);
 }
 
-static struct command_result *check_lost_peer(void *unused)
+static struct command_result *check_lost_peer(struct command *cmd)
 {
 	struct out_req *req;
-	req = jsonrpc_request_start(plugin, NULL, "listpeerchannels",
-					after_listpeerchannels,
-					&forward_error, NULL);
+	req = jsonrpc_request_start(plugin, cmd, "listpeerchannels",
+				    after_listpeerchannels,
+				    &forward_error, NULL);
 
 	return send_outreq(plugin, req);
 }
 
-static void do_check_lost_peer (void *unused)
+static struct command_result *do_check_lost_peer(struct command *cmd, void *unused)
 {
 
 	/* Set to NULL when already in progress. */
 	lost_state_timer = NULL;
 
-	if (recovery) {
-		return;
+	if (!recovery) {
```

Comment:
It should kill this timer only when it has already entered recovery mode.

Reply:
Thanks, fixed!